### PR TITLE
Skip check tests on host builds when IREE_BUILD_COMPILER is OFF.

### PR DESCRIPTION
**Pull request by ScottTodd**
I misconfigured this as part of https://github.com/google/iree/pull/4584, leading to issues spotted on https://github.com/google/iree/issues/4617. The flatcc cli is still required for the runtime, which is what I originally filed that issue for, but compiler tools like `iree-translate` should not be.

See also https://github.com/google/iree/blob/7d0108f0d565956c51c446dd52062fa1e50ca4e3/build_tools/cmake/iree_lit_test.cmake#L39-L43

### CODE REVIEW

File: build_tools/cmake/iree_check_test.cmake [hunk 1]
```diff
@@ -36,6 +36,12 @@ function(iree_check_test)
     return()
   endif()
 
+  # On the host, only build check tests if the compiler is enabled.
+  # When cross compiling, assume the host tools were already built.
+  if(NOT IREE_BUILD_COMPILER AND NOT CMAKE_CROSSCOMPILING)
```
Comment (GMNGeoffrey) [inline build_tools/cmake/iree_check_test.cmake:41]: Ok it took me a while to figure out how the comment corresponded to the if here. What if I want to use installed binaries while testing on the host? Basically, I think cross compiling doesn't have much to do with it here, so this feels like not quite the right way to check it. That said, that wasn't possible with our old cross-compiling either, so I'm fine with this workaround for now. But please tag an issue to clean this up. I think we might want something like `IREE_USE_INSTALLED_HOST_TOOLS`?

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/cmake/iree_check_test.cmake (hunk 1 above)
Comment (ScottTodd) [reply build_tools/cmake/iree_check_test.cmake:41]: Yeah, I think adding another cmake option makes the most sense here longer term. We could do something like set that new option on/off by default when cross compiling.

Comment (ScottTodd) [reply build_tools/cmake/iree_check_test.cmake:54]: Added way more detail to the comment.

Comment (GMNGeoffrey) [reply build_tools/cmake/iree_check_test.cmake:54]: Thanks :-) Could you please link an issue?

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/cmake/iree_check_test.cmake (hunk 1 above)
Comment (ScottTodd) [reply build_tools/cmake/iree_check_test.cmake:55]: Filed and linked to https://github.com/google/iree/issues/4662

**Pull request merged**